Zilon.Core/Zilon.BotEnvironment/AutoplayEngine.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Extensions.DependencyInjection;
+
+using Zilon.Bot.Players;
+using Zilon.Bot.Sdk;
+using Zilon.Core.Scoring;
+using Zilon.Core.Tactics;
+using Zilon.Emulation.Common;
+
+namespace Zilon.BotEnvironment
+{
+    class AutoplayEngine : AutoplayEngineBase<IPluggableActorTaskSource>
+    {
+        private const string SCORE_FILE_PATH = "bot-scores";
+        private const int BOT_EXCEPTION_LIMIT = 3;
+        private const int ENVIRONMENT_EXCEPTION_LIMIT = 3;
+
+        private readonly Startup _startup;
+        private readonly string scoreFilePreffix;
+
+        int botExceptionCount = 0;
+        int envExceptionCount = 0;
+
+        public AutoplayEngine(Startup startup,
+            BotSettings botSettings,
+            string scoreFilePreffix
+            ) : base(botSettings)
+        {
+            _startup = startup;
+            this.scoreFilePreffix = scoreFilePreffix;
+        }
+
+        protected override void CatchActorTaskExecutionException(ActorTaskExecutionException exception)
+        {
+            AppendException(exception, scoreFilePreffix);
+
+            var monsterActorTaskSource = ServiceScope.ServiceProvider.GetRequiredService<MonsterBotActorTaskSource>();
+            if (exception.ActorTaskSource != monsterActorTaskSource)
+            {
+                botExceptionCount++;
+
+                if (botExceptionCount >= BOT_EXCEPTION_LIMIT)
+                {
+                    AppendFail(ServiceScope.ServiceProvider, scoreFilePreffix);
+                    throw exception;
+                }
+            }
+            else
+            {
+                envExceptionCount++;
+                CheckEnvExceptions(envExceptionCount, exception);
+                Console.WriteLine($"[.] {exception.Message}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5028977)